src/main/java/seedu/contax/ui/onboarding/Overlay.java

```diff
@@ -64,54 +63,110 @@ public void showOnly(DoubleProperty boundX, DoubleProperty boundY,
+        overlay.minWidthProperty().bind(property);
+        overlay.maxWidthProperty().bind(property);
+    }
 
-        topOverlay.setStyle("-fx-background-color: rgb(0, 0, 0, 0.5)");
-        bottomOverlay.setStyle("-fx-background-color: rgb(0, 0, 0, 0.5)");
-        topOverlay.layoutXProperty().bind(boundX);
-        topOverlay.layoutYProperty().bind(new SimpleDoubleProperty(0));
-        topOverlay.minHeightProperty().bind(boundY);
-        topOverlay.minWidthProperty().bind(parentWidth);
-        topOverlay.maxHeightProperty().bind(boundY);
-        topOverlay.maxWidthProperty().bind(parentWidth);
-
-
-        bottomOverlay.layoutXProperty().bind(boundX);
-        bottomOverlay.layoutYProperty().bind(boundY.add(height));
-        bottomOverlay.minHeightProperty().bind(parentHeight.subtract(boundY.add(height)));
-        bottomOverlay.minWidthProperty().bind(parentWidth);
-        bottomOverlay.maxHeightProperty().bind(parentHeight.subtract(boundY.add(height)));
-        bottomOverlay.maxWidthProperty().bind(parentWidth);
+    /**
+     * Sets height of given overlay with the given ReadOnlyDoubleProperty.
+     * @param overlay overlay to set.
+     * @param property property to bind with.
+     */
+    private void setHeight(Pane overlay, ReadOnlyDoubleProperty property) {
+        overlay.minHeightProperty().bind(property);
+        overlay.maxHeightProperty().bind(property);
+    }
+
+    /**
+     * Sets height of given overlay with the given DoubleBinding.
+     * @param overlay overlay to set.
+     * @param property property to bind with.
+     */
+    private void setHeight(Pane overlay, DoubleBinding property) {
+        overlay.minHeightProperty().bind(property);
+        overlay.maxHeightProperty().bind(property);
+    }
+
+    /**
+     * Sets the given overlay with the given x and y ReadOnlyDoubleProperty.
+     * @param overlay overlay to set.
+     * @param x property to bind x with
+     * @param y property to bind y with
+     */
+    private void setLayout(Pane overlay, ReadOnlyDoubleProperty x, ReadOnlyDoubleProperty y) {
+        overlay.layoutXProperty().bind(x);
+        overlay.layoutYProperty().bind(y);
+    }
+
+    /**
+     * Sets the given overlay with the given x ReadOnlyDoublePropery and y DoubleBinding.
+     * @param overlay overlay to set.
+     * @param x property to bind x with
+     * @param y property to bind y with
+     */
+    private void setLayout(Pane overlay, ReadOnlyDoubleProperty x, DoubleBinding y) {
+        overlay.layoutXProperty().bind(x);
+        overlay.layoutYProperty().bind(y);
     }
 
     /**
-     * Show all overlays
+     * Shows the given overlay
+     * @param overlay
      */
-    private void showAll() {
-        topOverlay.setVisible(true);
-        bottomOverlay.setVisible(true);
+    private void showOverlay(Pane overlay) {
+        overlay.setVisible(true);
+        overlay.setStyle("-fx-background-color: rgb(0, 0, 0, 0.5)");
```

Comment:
Can you check if it should be `rgba` instead because you are also specifying the alpha value.

Reply:
Thanks for the catch. I think rgb works too but will update it to rgba